### ISSUE TITLE
[Bug]: Variable Setting Not Saving

### ISSUE THREAD

**Issue by avegel** [jasp-stats/jasp-issues#2648]
### JASP Version

0.18.3

### JASP Module

Other

### What OS are you seeing the problem on?

macOS Intel

### Bug Description

For both csv imported data and copied in data, JASP is not saving variable settings. The cariable settings are being reset after the JASP file is loaded.

Summary of my trouble shooting (**X** = problem persists / **O** = problem does not occur):
1. PC restart -> **X**
2. JASP reinstall -> **X**
3. Import data from csv file -> **X**
4. Copy data from csv -> **X**
5. Copy data from excel -> **X**
6. Change variable settings on old file (appx 4 months old, not sure if imported csv, copied in, or manually added) -> **O**
7. Data is manually entered in -> **O**

### Expected Behaviour

JASP files should retain variable settings regardless of how the data in imported or entered.

![Screenshot 2024-03-26 at 1 32 35 PM](https://github.com/jasp-stats/jasp-issues/assets/20810391/e94a1268-db2e-4657-8c72-0c9df8713c9e)

### Steps to Reproduce

1. Import data into by csv file into JASP, change variables, save, reopen.
2. Copy and paste data into into JASP from a csv or Excel file, change variables, save, reopen.
...

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if possible.
- [X] I have included a JASP file (zipped) or data file that causes the crash/bug, if applicable.
- [X] I have accurately described the bug, and steps to reproduce it.

**Comment by tomtomme** [comment]
CSVs cannot save those settings.
Did you try saving as a .jasp file? That should work.

**Comment by avegel** [comment]
Thomas,

Thanks for your response.

Sorry I wasn't clear about that. For numbers 3-5, all of these steps were
tested after saving the JASP file and reopening it. The problem persists. I
have attached one dataset as an example. The variable settings reset on
each reopen even after saving.

JASP is practically unusable if it cannot retain variable settings in large
datasets.


   1. PC restart -> *X*
   2. JASP reinstall -> *X*
   3. *Import data from csv file -> X*
   4. *Copy data from csv -> X*
   5. *Copy data from excel -> X*
   6. Change variable settings on old file (appx 4 months old, not sure if
   imported csv, copied in, or manually added) -> *O*
   7. Data is manually entered in -> *O*



Anton Vegel
PhD Student, Applied Linguistics, Georgia State University
MA, TESL, Kent State University
BA, TESL, Kent State University
Certificate, TEFL, Kent State University

**Comment by tomtomme** [comment]
Attaching the file via mail does not work. Pls rename the file to zip and drag and drop it here on the github website. I can check then tomorow maybe


And some questions.

- do you use a non english locale?
- whats your exact version of macOS?

**Comment by avegel** [comment]
[Archive.zip](https://github.com/jasp-stats/jasp-issues/files/14816378/Archive.zip)


Here's are test files (csv and JASP). I made sure to clear formatting of the csv before importing just in case.

**Comment by avegel** [comment]
1. These files are coming from a Japanese research partner. But wouldn't saving the file under csv UTF-8 solve any problems with that? 
2. I am using macOS SOnoma Version 14.4.1
3. All files are local (not called from usb or cloud).

**Comment by tomtomme** [comment]
Really strange

a) I cannot reproduce with your uploaded .jasp file. I changed some variables from nominal to scale or from ordinal to nominal etc. I saved the .jasp, reopened, and all changes were saved.
b) However when I made a new file and copy pasted the data from excel, my changes were not always saved. This was however not reproducible all the time. Sometimes it worked, sometimes not. Maybe I was to fast to close JASP after saving?

All this i on linux flatpak however.
Maybe someone from the team with macOS intel machine can try?

Also: Try saving under a different file name and file location / folder.

**Comment by boutinb** [comment]
Sorry @avegel I cannot reproduce this problem, but I'm not sure to understand what is your reproduction path: can you give step by step, with the Raw Test file you gave, a reproduction path of the issue?

**Comment by avegel** [comment]
@boutinb It's great to hear that the bug isn't affecting everyone. 

Here's a short video that shows the problem:
https://github.com/jasp-stats/jasp-issues/assets/20810391/1210a92f-f298-4bdf-845d-b685d5606d7c

Here's the summary:
1. Import csv file into JASP, 2. change variable settings, 3. load an analysis, 4. _variable settings reset_, 5. change variable settings, 6. save the JASP file, 7. close JASP, reopen JASP, 8. _variable settings reset_, change variable settings, resave the JASP file, reopen JASP, _variable settings reset_

In short, variable settings reset when loading an analysis or when reopening a JASP file.

**Comment by boutinb** [comment]
I understand now your problem, and I can reproduce it. Quite strange. But fortunately with the development version, this seems to be solved (we have changed many things in the code handling the type change). Can you try it: https://static.jasp-stats.org/Nightlies/macOS-x86_64/JASP-nightly-development-c8c4acf1dcb12480b97fe7e07583b903366768df.dmg
This is of course just a development version, which is not fully tested! We think to have a release over 1 month.

**Comment by avegel** [comment]
Thanks!

**Comment by tomtomme** [comment]
@avegel 
Did you test the new build? Did it work?
Or do you want to wait for the final 0.19 release?

**Comment by avegel** [comment]
The new build seems to work!

Much thanks to the quick attention to the bug report.

I'm a PhD student at Georgia State University, and this project is for a
term paper and relatively low-risk. I will recheck all tests once 0.19 is
officially released. I think the new build (pre-release) will do for now.



Anton Vegel
PhD Student, Applied Linguistics, Georgia State University
MA, TESL, Kent State University
BA, TESL, Kent State University
Certificate, TEFL, Kent State University
